codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/localservice/DafnyLocalServiceGenerator.java

```diff
@@ -418,9 +471,44 @@ void generateShim() {
             final var inputShape = model.expectShape(
               operationShape.getInputShape()
             );
-            final var outputShape = model.expectShape(
+            var outputShape = model.expectShape(
               operationShape.getOutputShape()
             );
+            var isMemberShapePointable = true;
+            String toDafnyConvMethodName =
+              SmithyNameResolver.getToDafnyMethodName(outputShape, "");
+            if (outputShape.hasTrait(PositionalTrait.class)) {
+              // Shape with positional shape MUST have only one membershape which is always required.
```

Comment:
Based on our offline discussion I think this is not entirely true. It's not exactly required from smithy pov but behaves like one from implementation pov.